rbtree.go

```diff
@@ -133,7 +133,7 @@ func (t *Rbtree) rightRotate(x *Node) {
 	x.Parent = y
 }
 
-func (t *Rbtree) insert(z *Node) {
+func (t *Rbtree) insert(z *Node) Item {
```

Comment:
If we must return a value here, I prefer it be a `Node`.  Let those exported methods to deal with the things like `Item`, WDYT